sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -30,7 +29,18 @@ import org.apache.spark.unsafe.types.UTF8String
  * for classes whose fields are entirely defined by constructor params but should not be
  * case classes.
  */
-private[sql] trait DefinedByConstructorParams
+private[sql] trait DefinedByConstructorParams {
+
+  /**
+   * Return the values of this instance's constructor parameters.
+   */
+  private[sql] def constructorParams: Seq[AnyRef] = {
```

Comment:
I kind of prefer not having this here and put it in show itself so we don't introduce a public API in Java for every class that extends this.